### PR TITLE
1109 modal footer bg

### LINKED ISSUE #1109: [Enhancement] Theming on Footer (White/Background Color)
**Is your enhancement request related to a problem? Please describe.**
Right now our footer is always white, but we have components that are not visible when placed on a white background, like the pincode component in team1, which has a white background.

**Describe the solution you'd like**
We would like to be able to change the color on our footer, based on some kind of theming. We do not want to allow all colors, just some. 

In the UX team we have decided that it's either White or Background color, which could be translated to some kind of theme, like "White/Light" or we could only allow white and background color.

See also: 
Skærmbillede 2020-08-10 kl  12 23 09

https://zpl.io/aNXwPGk

### PR DESCRIPTION
**Pull request by peterlindholm**
This PR closes #1109 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: 1109

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts [hunk 1]
```diff
@@ -206,6 +220,32 @@ describe('ModalWrapperComponent', () => {
...
+  //   });
+
+  //   it('should default to white background on embedded footer', () => {
+  //     const kirbyModalFooter = spectator.element.querySelector(':scope > kirby-modal-footer');
+  //     expect(kirbyModalFooter).toHaveComputedStyle({
+  //       'background-color': getColor('white'),
+  //     });
+  //   });
+  // });
+
```
Comment (peterlindholm) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts:248]: I'm missing something here...I get this error:
Expected [background-color] of KIRBY-MODAL-FOOTER 'rgba(0, 0, 0, 0)' to equal 'rgb(255, 255, 255)' (white)

File: libs/designsystem/src/lib/components/modal/footer/modal-footer.component.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
-<ion-footer>
+<ion-footer [class.light-bg]="themeColor && themeColor === 'light'">
```
Comment (hafstad) [inline libs/designsystem/src/lib/components/modal/footer/modal-footer.component.html:1]: Consider using the theme-color directive instead, so we don't need theme-color logic inside individually components like avatar, badge, card etc.

File: libs/designsystem/testing-base/src/lib/components/mock.modal-footer.component.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ import { ModalFooterComponent } from '@kirbydesign/designsystem';
   ],
 })
 export class MockModalFooterComponent {
+  @Input() themeColor: 'white' | 'light';
```
Comment (hafstad) [inline libs/designsystem/testing-base/src/lib/components/mock.modal-footer.component.ts:17]: If theme-color directive is used input aren't necessary

File: libs/designsystem/src/lib/components/modal/footer/modal-footer.component.ts [hunk 4]
```diff
@@ -6,6 +6,8 @@ import { Component, Input, HostBinding } from '@angular/core';
   styleUrls: ['./modal-footer.component.scss'],
 })
 export class ModalFooterComponent {
+  @Input()
+  themeColor?: 'white' | 'light';
```
Comment (hafstad) [inline libs/designsystem/src/lib/components/modal/footer/modal-footer.component.ts:10]: If theme-color directive is used input aren't necessary

**Review by hafstad — APPROVED** [review]

**Pull request merged**